### DISCUSSION TITLE
How to fit object to window size?

### DISCUSSION THREAD

**Discussion by Sriram1023** [pyvista/pyvista#3378] [Q&A]
Hi,

I would like to view an obj using the zx view and set zoom such that it fills the render window.

I have tried using plotter.camera.zoom('tight'). however, this results in object getting rotates to its minimal axis and getting zoomed.
is there a way to do tight without rotating the object/ or programatically finding the optimal zoom level? from object bounds or something?

**Comment by geostar1024** [comment]
I was having the same problem, but at least now a solution exists (in 2026)! At some point a method `tight()` was added to the camera object, which takes an optional viewing plane. So `pl.camera.tight(view='zx')` will do the job.

https://docs.pyvista.org/api/core/_autosummary/pyvista.camera.tight
